.github/workflows/codecoverage.yml

```diff
@@ -0,0 +1,58 @@
+name: Code Coverage (Apollo tests)
+on:
+  schedule:
+    - cron: "0 0 * * 6"
+
+jobs:
+  build:
+    name: codecoverage
+    runs-on: ubuntu-18.04
+    strategy:
+      fail-fast: false
+    steps:
+      - name: Cleanup pre-installed tools
+        run: |
+          # This is a fix for https://github.com/actions/virtual-environments/issues/1918
+          sudo rm -rf /usr/share/dotnet
+          sudo rm -rf /opt/ghc
+          sudo rm -rf "/usr/local/share/boost"
+          sudo rm -rf "$AGENT_TOOLSDIRECTORY"
+      - name: Checkout repository code
+        uses: actions/checkout@v2
+      - name: Create artifact directory
+        run: mkdir -p ${{ github.workspace }}/artifact
+
+      - name: Build and test
+        id: build_test
+        run: |
+          script -q -e -c "make pull"
+          sudo df -h
+          script -q -e -c "CONCORD_BFT_CMAKE_BUILD_TYPE=Debug CONCORD_BFT_CMAKE_CODECOVERAGE=TRUE make build" \
```

Comment:
Does it need to be the `Debug` build? Ideally (if even possible), I'd suggest prioritizing the `Release` build for code coverage calculation.

Reply:
@teoparvanov , I think it should be Debug because if code coverage is on, the optimization is turned off:
```
string(APPEND CMAKE_CXX_FLAGS " -g -O0")
```

Reply:
Is the code coverage based on LLVM? If so, I guess need to set the compiler to clang explicitly.

Reply:
@f-squirrel Yes it is LLVm based.

Reply:
clang specific flags have been added.